pkg/logging/controller/logging/clusterlogging.go

```diff
@@ -0,0 +1,92 @@
+package logging
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	loggingconfig "github.com/rancher/rancher/pkg/logging/config"
+	"github.com/rancher/rancher/pkg/logging/generator"
+	"github.com/rancher/rancher/pkg/logging/utils"
+)
+
+type ClusterLoggingSyncer struct {
+	management           config.ManagementContext
+	clusterLoggingClient v3.ClusterLoggingInterface
+	projectLoggingClient v3.ProjectLoggingInterface
+	K8sClient            kubernetes.Interface
+}
+
+func RegisterClusterLogging(cluster *config.ClusterContext) {
+	clusterloggingClient := cluster.Management.Management.ClusterLoggings("")
+	syncer := &ClusterLoggingSyncer{
+		management:           *cluster.Management,
+		K8sClient:            cluster.K8sClient,
```

Comment:
We should try avoiding using a generic API client as it's harder to track of what it is used for. We’ve kinda agreed on using `ObjectLister` for accessing object in cache, and `Objects` for the client. So example for cluster would be `ClusterLister` and `Clusters`